test/e2e/e2e_test.go

```diff
@@ -240,6 +233,15 @@ func beforeE2ETest(t *testing.T) *testState {
 		}
 	}
 
+	// If no namespace is set to run tests against, use the default namespace
+	if os.Getenv("NAMESPACE") != "" {
+		namespace := os.Getenv("NAMESPACE")
+		CreateNamespaces(ts.client1, namespace)
+		CreateNamespaces(ts.client2, namespace)
+	} else {
+		os.Setenv("NAMESPACE", "default")
```

Comment:
Rather than setting an environment variable here, and always reading from an environment variable, maybe we should set a field on the `TestState`?

Reply:
will do!